scrapy/core/engine.py

```diff
@@ -163,58 +179,84 @@ def pause(self) -> None:
+        requests."""
+        # Starts the processing of scheduled requests, as well as a periodic
+        # call to that processing method for scenarios where the scheduler
+        # reports having pending requests but returns none.
+        assert self._slot is not None  # typing
+        self._slot.nextcall.schedule()
+        self._slot.heartbeat.start(self._SLOT_HEARTBEAT_INTERVAL)
+
+        while self._start and self.spider:
+            await self._process_start_next()
+            if not self.needs_backout():
+                # Give room for the outcome of self._process_start_next() to be
+                # processed before continuing with the next iteration.
+                self._slot.nextcall.schedule()
+                await self._slot.nextcall.wait()
+
+    def _start_scheduled_requests(self) -> None:
+        if self._slot is None or self._slot.closing is not None or self.paused:
             return
 
-        while (
-            not self._needs_backout()
-            and self._next_request_from_scheduler() is not None
-        ):
-            pass
-
-        if self.slot.start_requests is not None and not self._needs_backout():
-            try:
-                request_or_item = next(self.slot.start_requests)
-            except StopIteration:
-                self.slot.start_requests = None
-            except Exception:
-                self.slot.start_requests = None
-                logger.error(
-                    "Error while obtaining start requests",
-                    exc_info=True,
-                    extra={"spider": self.spider},
-                )
-            else:
-                if isinstance(request_or_item, Request):
-                    self.crawl(request_or_item)
-                else:
-                    self.scraper.start_itemproc(request_or_item, response=None)
+        while not self.needs_backout():
+            if self._start_scheduled_request() is None:
+                break
 
-        if self.spider_is_idle() and self.slot.close_if_idle:
+        if self.spider_is_idle() and self._slot.close_if_idle:
             self._spider_idle()
 
-    def _needs_backout(self) -> bool:
-        assert self.slot is not None  # typing
+    def needs_backout(self) -> bool:
+        """Returns ``True`` if no more requests can be sent at the moment, or
+        ``False`` otherwise.
+
+        See :ref:`start-requests-lazy` for an example.
+        """
+        assert self._slot is not None  # typing
         assert self.scraper.slot is not None  # typing
         return (
             not self.running
-            or bool(self.slot.closing)
+            or bool(self._slot.closing)
             or self.downloader.needs_backout()
             or self.scraper.slot.needs_backout()
         )
 
-    def _next_request_from_scheduler(self) -> Deferred[None] | None:
-        assert self.slot is not None  # typing
+    def _start_scheduled_request(self) -> Deferred[None] | None:
```

Comment:
The returned deferred is never used, but it was the same with `_next_request_from_scheduler()`.

I wonder if it's saved somewhere deep inside the code or is just garbage collected at a random time.

Reply:
Though it may be a correct way to use deferreds?

Reply:
Yes, I think you do not need to await deferreds. I think what’s an issue is when you define a deferred but never trigger or cancel it, but it is fine not to await deferreds.